### DISCUSSION TITLE
When will GitHub Packages add support for GitHub Apps?

### DISCUSSION THREAD

**Discussion by dgholz** [community/community#34324] [Packages]
I want to generate an OAuth token (like from `gh auth`) via our setup tools when onboarding new devs, so they don't need to generate a PAT (and remember to expire it, rotate it, etc.) to install our private packages from GitHub Packages.

I know that GitHub Packages only supports PATs through its API and the language-specific registries. I see that it's not very obvious, a lot of people have asked about this (#33202, #28039, #34084).

When will GitHub Packages support tokens other than PATs? That is, GitHub App tokens and OAuth App tokens.

**Comment by chriscarpenter12** [comment]
This seems pretty shortsighted not being able to interact with Packages from an App. What's more confusing is that on the permission page for Apps there's a setting for accessing Packages which implied to me it had access

![image](https://user-images.githubusercontent.com/458486/205137241-1033fdc5-601e-4eaf-b8a0-2cdbb2787127.png)

But I too get permission denied when trying to push a package using an App `ghs` token

```
npm ERR! code E403
npm ERR! 403 403 Forbidden - PUT https://npm.pkg.github.com/@org%2fpackage-name - Permission permission_denied: installation not allowed to Create organization package
npm ERR! 403 In most cases, you or one of your dependencies are requesting
npm ERR! 403 a package version that is forbidden by your security policy, or
npm ERR! 403 on a server you do not have access to.
```

**Reply by simonoff** [reply]
Yes, but GitHub focusing more in Copilot than on this "stupid" feature

**Reply by Drowze** [reply]
Also if you go to your package settings page at https://github.com/orgs/ORG/packages/rubygems/PACKAGE/settings, you'll find an access setting as such:
![Screenshot 2023-09-22 at 12 42 54](https://github.com/community/community/assets/9031589/339800a0-02d7-46c1-98ca-6cf5d0ed747b)

...but adding repositories to that setting really did nothing. I didn't find any way that I can download packages (tried with Ruby packages) using the GITHUB_TOKEN on my GH Actions workflow (which I suppose is similar if not the same as a GitHub App-generated user token).

This is really unfortunate... I don't see any solution besides service accounts with Personal Access Tokens 😢

**Comment by uonehu** [comment]
تنظیمات

**Reply by uonehu** [reply]
صورت حساب

**Reply by uonehu** [reply]
[]()

**Comment by mr-pinzhang** [comment]
It seems like a very practical feature. Hopefully, GH team can have it done very soon.

**Comment by fsdh-breakglass-01** [comment]
Would love to have this option.

**Comment by cmhac** [comment]
This limitation is also causing me headaches today in November 2024

**Comment by mat007** [comment]
Yeah, pretty annoying to keep bot users around just for that reason…

**Comment by iocmet** [comment]
.... Currently only way to access repository packages is classic PAT?

**Reply by boomshakar** [reply]
You can use `secrets.GITHUB_TOKEN` too though

**Reply by iocmet** [reply]
I know, i meant not from actions though

**Comment by john-funk** [comment]
One really has to search the community discussions before trying for hours to get stuff working that seem obvious, but that GitHub decided to just not care about for years...

**Comment by ct-alex-au** [comment]
Personally I would like to use GitHub Apps with short-lived user OAuth token (device flow) to manage package registries auth instead of letting users to generate tons unexpired PATs

**Comment by willrichboa** [comment]
In the modern security environment please give me GitHub App's to manage publishing my NPM packages.

GitHub Apps are generally preferred over Personal Access Tokens because they offer enhanced security, fine-grained permissions, and can act independently of a user. In contrast, PATs are tied to individual users and require management of their lifecycle.
